### ISSUE TITLE
Display missing count at top of group's page + progress bars

### ISSUE THREAD

**Issue by MugUK** [atari-legend/atari-legend#93]
E.g. Damien's 666 page - https://menus.atarilegend.com/menusets/50

It says at the top that this page contains:

Damien (666) contains 2 menus and 2 disks.

But both menus are missing. So do we keep a total of the number of disk images missing from a group, so that the top could be changed to:

Damien (666) contains 2 menus and 2 disks. <Currently missing 2 menus>.

or if it's a perfect set:

<100% Complete>

**Comment by MugUK** [comment]
I don't know if this is linked but on the Deep Thought's page (that apostrophe hurts my brain!) it was 1 menu, with the full list of contents and a screenshot and it is labelled as intact .. but there is no download is available. 

Should we include a reason why?

**Comment by Brume-AL** [comment]
Good point about Deep Thought's, looks like apostrophe causes an issue on the site.
Same goes with Cynix Dizzy's Excellent Adventures

https://menus.atarilegend.com/menusets/47

**Comment by MugUK** [comment]
So an apostrophe in a menu group's title breaks the download?

Is that also why Flame of Finland/Superior menus aren't downloadable?

**Comment by nguillaumin** [comment]
Thanks for the feedback guys, I'm so glad I asked people to test :smile: There was a bug in the import routine whenever the menu set name has a special character on it. It's because the menu set name is used as a folder name (e.g. `Cynix_Dizzy_s_Excellent_Adventures`) and I was not using the same escape rules as Stonish, so my import script did not find the ZIP files.

Note for myself: This is the SQL query to use to find such errors (menus that are intact but don't have a dump):

```sql
SELECT
    *
FROM
    menu_disks
LEFT JOIN menus ON menu_disks.menu_id = menus.id
LEFT JOIN menu_sets ON menus.menu_set_id = menu_sets.id
WHERE
    menu_disk_condition_id = 4 AND menu_disk_dump_id IS NULL
```

After fixing the problem in the script I still see some menus like that:
- Essays Favoritdisk 1, 2, 3
- Flame Of Finland/Superior 118
- Pompey Pirates 35, 36
- Regression Therapy 1

As far as I can tell this is an error in the Stonish DB (Condition is 4 = Intact, but there's no download):

![2](https://user-images.githubusercontent.com/762512/108420462-2801bb00-7234-11eb-956e-8fc78fc5fb22.png)

@Brume-AL is it something you can fix, so that it will be good next time I import? I can also do it if you want.

**Note:** I will need to do a re-import on the test site, so the downloads will still be missing for now.

I have also added a count of missing disks and completion percentage:

![3](https://user-images.githubusercontent.com/762512/108423729-8335ac80-7238-11eb-84c5-f1d39743409e.png)

![4](https://user-images.githubusercontent.com/762512/108423731-83ce4300-7238-11eb-909e-b065a53380bc.png)

I thought maybe that would be an interesting to display on the list of menu sets. Maybe we can "fill" the menuset name bar depending how complete the set is? A quick mockup for a set that is 90% complete for example:

![5](https://user-images.githubusercontent.com/762512/108424059-f808e680-7238-11eb-84b5-1d60741a154b.png)

**Comment by Brume-AL** [comment]
Thanks for reporting the issue.
I wrote long time a small php page: when a menu has a file attached but it's marked as missing, the page displays the menu. But I should have taken the opposite case into consideration, too.
It's fixed on Stonish website now.

**Comment by Brume-AL** [comment]
I like the bar idea but I'm not a fan of the grey color inside the tile.
Maybe could we use something like that and adapt it to our colors, I don't know?

![Curseur](https://user-images.githubusercontent.com/12120244/108426941-b5e1a400-723c-11eb-817b-fe153843494d.jpg)

**Comment by MugUK** [comment]
I'm not sure we can do "The set is 100% complete", especially with A Clockwork Orange menus as no-one knows how many exist. We got an extra two purely by luck when I bought that Atari ST bundle with all of those 5.25" disks :)

Maybe reword it to say "Our set is 100% complete" for the menu groups where we don't know the full range of menus? Or is that being too picky?

**Comment by MugUK** [comment]
Also, and this maybe a feature update, if there's only 1 disk missing, then it shouldn't say "1 disks missing" like it does on Fuzion menu page. https://menus.atarilegend.com/menusets/94?page=2#menudisk-2010

Or change the message to say "x disk(s) missing".

**Comment by nguillaumin** [comment]
I have changed the wording to "All known disks are available, our set is 100% complete!". I also fixed the singular/plural for "X disks missing". (It should get deployed in a few minutes).

**Comment by MugUK** [comment]
Perfect.

**Comment by nguillaumin** [comment]
Here's an attempt at progress bars:

![2](https://user-images.githubusercontent.com/762512/108599816-b17bcf00-7393-11eb-82f2-345e2ca113db.png)

I think the colors are too distracting? Perhaps just using a single color:

![3](https://user-images.githubusercontent.com/762512/108599821-b50f5600-7393-11eb-96e0-fad66f3f3556.png)

Let me know what you think?

**Comment by nguillaumin** [comment]
Another variation:

![4](https://user-images.githubusercontent.com/762512/108601735-66b38480-739e-11eb-8c2e-2c71f769302c.png)

**Comment by Brume-AL** [comment]
The Last one is very cool! Love it, I vote for this one 👍👍👍

**Comment by Brume-AL** [comment]
Just a small request: could you try to use a biggest font for the names of the bands? Or something written in a different font like this one, maybe?

![Capture d’écran 2021-02-20 173800](https://user-images.githubusercontent.com/12120244/108602505-88166f80-73a2-11eb-9fb0-168fd184cd6a.jpg)

**Comment by nguillaumin** [comment]
Using the Audiowide font doesn't work well I think. We reserve it really just for headers, and it feels a bit "too much" to me:

![7](https://user-images.githubusercontent.com/762512/108602844-a9785b00-73a4-11eb-875a-4160cea701c2.png)

I have increased the font size a bit on the test site, have a look.

Let me know what you think? If you think that works with the Audiowide font, then I can change it.

**Comment by MugUK** [comment]
Fuzio still has this at the top, I guess due to rounding up?

"1 disk missing (100% complete). "

**Comment by nguillaumin** [comment]
Thanks, rounding indeed. I fixed it, but I can't find a way in PHP to omit the decimals if they are zero, so it will show "20.0% complete" (I would have liked "20%", but if there's a decimal like "86.5" it should be shown).

**Comment by stgraveyard** [comment]
Nico, 

May I do one suggestion, just to test it. The progress bar, I love it, but overall this page now has a bit too much legacy blue. So could you make the progress bar from black, dark grey (#6e6e6e) to light gray (#cccccc). I just like to see it if it looks nice to break all the blue/green a bit...

**Comment by nguillaumin** [comment]
There is it:

![5](https://user-images.githubusercontent.com/762512/108630247-1dc60380-7464-11eb-8430-aca174246716.png)

I'm not sure what to think of it. We don't use grey that much elsewhere so it feels a bit out of place.

Something else I tried (closer to the initial mockup I proposed):

![4](https://user-images.githubusercontent.com/762512/108630269-37674b00-7464-11eb-8a08-9961e24b0b05.png)

**Comment by nguillaumin** [comment]
I have deployed the first version, if you want to check it out.

**Comment by stgraveyard** [comment]
Thanks a lot Nico. I personally like the gray tones a bit more, as it is a bit variety, but I let you, Bruno and the other decide. I also like the last one. What I don't like is the centering of the text : 

![image](https://user-images.githubusercontent.com/4339294/108640031-cc376c00-7497-11eb-96c9-831cd927ea6d.png)

It should be centered in the width of the whole bar, not just the availabilty color. Don't know if this is possible. And taking this into account, probably the first one, with the all blue colors fits best, as the white font is always visible in that color scheme.

**Comment by stgraveyard** [comment]
Also, I don't know what changed, maybe the width, or maybe the font size, but on my screen it looks like this : 

![image](https://user-images.githubusercontent.com/4339294/108640228-c2facf00-7498-11eb-9fc0-ce036581af3f.png)

It looks a bit cramped to me. I don't know if I'm dreaming, but I thought it looked a bit cleaner before?

**Comment by Brume-AL** [comment]
I don't have such an issue. Do you use any scaling setup?
Here are both captures on my PC:

- first monitor - 2560 x 1540 (scaling 125%)
![11](https://user-images.githubusercontent.com/12120244/108812829-1875cf80-75b0-11eb-8242-c871a476f75d.jpg)


-second monitor - 1920 x 1080 (no scaling)

![22](https://user-images.githubusercontent.com/12120244/108812920-3f340600-75b0-11eb-91b5-9a87332feb7f.jpg)

**Comment by nguillaumin** [comment]
It's because I increased the font size at Brume's request. I agree it's a bit cramped and I preferred before too 😉 but maybe it's just that we were used to the previous size.

**Comment by stgraveyard** [comment]
Yes, I just write down what I notice but ... in the end I think it is still all very nice, so we will leave it then ...

**Comment by stgraveyard** [comment]
I like it, only mild comment I have it the centering of the xx/xx discs. But I like the gray progress bar...

**Comment by stgraveyard** [comment]
You know, I'm just gonna say it. I'm checking the site more and more and this : 

![image](https://user-images.githubusercontent.com/4339294/109390336-5573f600-7911-11eb-87fd-87ecb3ff3664.png)

I still think I would like to see it with a smaller random screenshot of the group (adding more color) and a smaller font ... 

Sorry, I think it just looks different :-)

I will shut up now ... ;-)

**Comment by nguillaumin** [comment]
Unfortunately I can't really center the text inside the progress bar, partly because it would need to be half black (on the left) and half white (on the right, over the black background) which is not possible.

I have tweaked the page a bit. I move the number of disk above the progress bar. I added a text + icon for complete sets.

I also reduced the font-size again (Sorry @Brume-AL !), I agree with you I think it works better that way. Less menu names are causing a new line break when they're smaller and it looks better.

**Comment by stgraveyard** [comment]
Yes Nico, very slick looking, love it! 
Only thing missing are the menu disk images ... but I guess this is a nice compromise !

**Comment by MugUK** [comment]
Like it.

**Comment by nguillaumin** [comment]
Ok. I will not add the screenshots for now, I really think it doesn't work that well :wink:
